### ISSUE TITLE
sklearn

### ISSUE THREAD

**Issue by yufang67** [yufang67/machine-learning-articles#16]
# TL;DR
## API
- estimator
 - model.fit()
 - model.coef_ ,  model.labels_ , model.intercept_ , model.cluster_centers_ , model.inertia_
 - input X should be two dimension: X = [:,np.newaxis]

- predictor
 - model.predict()
 - model.predict_proba()
 - model.score(trainX, testX)
 - model.decision_function(testX)

- transformer (.fit() .transform())
 - LabelEncoder: 1d, LE.classes_
 - OrdinalEncoder: nd, OE.categories_ , 有顺序
 - OneHotEncoder: 2d
 - MinMaxScaler: 2d, X.reshape(-1,1)
 - StandardScaler: 2d

- meta estimator

 - ensemble.BaggingClassifier
 - ensemble.VotingClassifier
 - multiclass.OneVsOneClassifier
 - multiclass.OneVsRestClassifier
 - multioutput.MultiOutputClassifier
 - impute.SimpleImputer
 - model_selection.GridSearchCV
 - model_selection.RandomizedSearchCV
 - pipeline.Pipeline
![image](https://mmbiz.qpic.cn/mmbiz_png/e4kxNicDVcCFicrpeIAOcOJScf7vd9NuPFAiaicIbEo8ZxyiaoyzxqQKOeCDAFpPEibU9pLmZvDjNLJgImcX3ceNicDibw/640?wx_fmt=png&tp=webp&wxfrom=5&wx_lazy=1&wx_co=1)

![image](https://mmbiz.qpic.cn/mmbiz_png/e4kxNicDVcCFicrpeIAOcOJScf7vd9NuPF8sp1g9WLl1hdiaunrjztjK214TZa5joBibDDyRCey4cZu7deqwsoxRPg/640?wx_fmt=png&tp=webp&wxfrom=5&wx_lazy=1&wx_co=1)

# Ref
[非常详细的sklearn介绍](https://mp.weixin.qq.com/s?__biz=MzU0MDQ1NjAzNg==&mid=2247489933&idx=1&sn=6ca1058440a9cf4c8e82a8c683965b1a&chksm=fb39bc86cc4e35909598ab5394ad21c81a2cff7e9d4b93889cd0f9a3aeb2d722f130a4194bf7&mpshare=1&scene=24&srcid=0909jsl0ZddLw1lXO1iKSsE2&sharer_sharetime=1599662878530&sharer_shareid=50db2e6276d3a235822e392fb4f1a098#rd)
